agent/metrics_test.go

```diff
@@ -420,3 +442,31 @@ func TestHTTPHandlers_AgentMetrics_CACertExpiry_Prometheus(t *testing.T) {
 	})
 
 }
+
+func TestMetricsOutput(t *testing.T) {
+	skipIfShortTesting(t)
+
+	t.Run("RPC calls with elapsed time below 1ms are reported as decimal", func(t *testing.T) {
+		metricsPrefix := "new_rpc_metrics_2"
+		allowRPCMetricRule := metricsPrefix + "." + strings.Join(middleware.OneTwelveRPCSummary[0].Name, ".")
+		hcl := fmt.Sprintf(`
+		telemetry = {
+			prometheus_retention_time = "5s"
+			disable_hostname = true
+			metrics_prefix = "%s"
+			prefix_filter = ["+%s"]
+		}
+		`, metricsPrefix, allowRPCMetricRule)
+
+		a := StartTestAgent(t, TestAgent{HCL: hcl})
+		defer a.Shutdown()
+
+		var out struct{}
+		_ = a.RPC("Status.Ping", struct{}{}, &out)
+
+		respRec := httptest.NewRecorder()
+		recordPromMetrics(t, a, respRec)
+
+		assertMetricsWithLabelHaveValueDifferentFrom(t, respRec, "method", "Status.Ping", "0")
```

Comment:
Thanks for following the testing framework we outlined!

Let's please move this check `assertMetricsWithLabel...` in [`TestAgentOneTwelve...`](https://github.com/hashicorp/consul/pull/12905/files#diff-472e516d540a393b43170efc09098871a521b8b8ed4788089412829311ff81c5L148) since this would be an improvement